js/parts/ColumnSeries.js

```diff
@@ -513,59 +547,81 @@ seriesType('column', 'line',
             height: h
         };
     },
+    /**
+     * Dirty the series that should be redrawn after addPoint(),
+     * point.remove() or similar.
+     *
+     * @private
+     * @function Highcharts.seriesTypes.column#dirtyTheSeries
+     * @return {void}
+     */
     dirtyTheSeries: function (series, point) {
         series.xAxis.series.forEach(function (otherSeries) {
             if (otherSeries.type === series.type &&
-                otherSeries.hasPointInX(point, otherSeries)) {
+                otherSeries.hasValueInX(point, otherSeries)) {
                 otherSeries.isDirty = true;
             }
         });
     },
-    findMinColumnWidth: function () {
+    /**
+     * Calculate the width of the narrowest column.
+     * When `series.ignoreNulls` is either `"normal"` or `"evenlySpaced"`,
+     * the `minColumnWidth` is calculated before first series.translate().
+     *
+     * @private
+     * @function Highcharts.seriesTypes.column#getMinColumnWidth
+     * @return {number}
+     */
+    getMinColumnWidth: function () {
         var series = this, chart = series.chart;
-        var colWidth;
+        var colWidth, minColWidth = Infinity;
         Series.prototype.translate.apply(series);
-        if (series.index === series.chart.series.length - 1) {
-            chart.series.forEach(function (series) {
-                if (series.type === 'column' ||
-                    series.type === 'columnrange' ||
-                    series.type === 'bar') {
-                    series.points.forEach(function (point) {
-                        colWidth = series
-                            .getColumnMetrics(point).width;
-                        if (chart.minColumnWidth === void 0 ||
-                            colWidth < chart.minColumnWidth) {
-                            chart.minColumnWidth = colWidth;
-                        }
-                    });
-                }
-            });
-        }
+        chart.series.forEach(function (series) {
+            // Make sure that this series is the last
+            // of column/columnrange/bar.
+            if ((series.type === 'column' ||
+                series.type === 'columnrange' ||
+                series.type === 'bar') && series.points) {
```

Comment:
Consider to use match with regexp like: 

```
series.type.match(/column|columnrange|bar/g)
```

Reply:
Don't check `series.type` string, instead use `series instanceof seriesTypes.column` (which will be sooner or later replaced anyway). Other series types, like `errorbar` should share this functionality too.

Reply:
@pawelfus the condition `series instanceof seriesTypes.column` is not true for `columnrange` series type which does not inherit from `column`.

Reply:
Aahh.. thanks for the explanation! FYI @TorsteinHonsi